### ISSUE TITLE
WebSocket connection error on prerelease2.triplea-game.org

### ISSUE THREAD

**Issue by RoiEXLab** [triplea-game/triplea#5998]
**How can the problem be recreated?**
- Open TripleA using latest master
- Click on "Play Online" and login with your credentials
- Get the error: "Failed to connect to: wss://prerelease2.triplea-game.org/lobby/games/listing-ws"

Not sure what the exact implications are, but the dialog is annoying ^^

**When the problem is fixed, what should have happened instead?**
Just the usual connect to lobby stuff

**Which Engine Version are you using?** lastest master

**Screenshots**
![Screenshot](https://user-images.githubusercontent.com/8350879/75779146-666b4500-5d59-11ea-8018-40296396331d.png)

**Comment by DanVanAtta** [comment]
The game listing WS notifies clients of changes in the lobby game listing
after their initial connection to the lobby, eg: game updates like player
joined or map changed

**Comment by DanVanAtta** [comment]
IIRC, we did not see this on prerelease and then either on the prerelease2 DNS entry or perhaps before then we had a regression that triggered this issue. We also do not see this problem on local. 

Bisecting this problem to find out when it started would be painful as we'd have to deploy to prerelease to repro. If some initial debugging can't reveal the problem, we may need to consider doing websocket communication over (non-secure) 'ws' instead of 'wss'

**Comment by RoiEXLab** [comment]
@DanVanAtta I actually don't think it's an TLS problem.

I got this stacktrace:
```
SEVERE: Websocket error
javax.net.ssl.SSLException: Socket closed
	at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:127)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:320)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:263)
	at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:258)
	at java.base/sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1313)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:839)
	at java.base/java.io.InputStream.read(InputStream.java:205)
	at org.java_websocket.client.WebSocketClient.run(WebSocketClient.java:424)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.SocketException: Socket closed
	at java.base/java.net.SocketInputStream.socketRead0(Native Method)
	at java.base/java.net.SocketInputStream.socketRead(SocketInputStream.java:115)
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:168)
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:448)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:68)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1103)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:823)
	... 3 more
```

Which indicates that the server decided to close the connection at some point, not some sort of TLS problems.

Because local works, I believe it must be an nginx configuration issue that is is somehow not letting the websocket request through and just blocking it.

**Comment by RoiEXLab** [comment]
@DanVanAtta I think I found the issue, haven't verified it but I'm pretty sure it's the problem:
The path `/lobby/games/listing-ws` is not configured as websocket endpoint in nginx.

https://github.com/triplea-game/triplea/blob/7439e25c1fef3a3f66ec33aeeb7502a798fe2748/infrastructure/ansible/roles/nginx/templates/etc_nginx_sites_enabled_default.j2#L50

We basically have 2 solutions for this problem:
1. We simply add a new rule for every websocket endpoint we create, either by duplicating most of the rules or by using a regex to match multiple paths [as described here](https://stackoverflow.com/a/35369570).
2. We just merge all the endpoints into a single one, used by all websocket endpoints. IIRC there was this problem where the websocket endpoint wouldn't care anyways.

Basically the first option can be done quickly, but becomes very ugly very fast once we start adding endpoints, whereas the second option might be a cleaner approach, but takes a little bit more work

**Comment by DanVanAtta** [comment]
@RoiEXLab good detective work.
I put up a PR to add the config: #6012 
IMO both options perhaps the way to go, (1) first for the problem remediation, then probably (2) as we update/fix overall design (ie: https://github.com/triplea-game/triplea/issues/5845)

**Comment by prastle** [comment]
Does this explain why no longer able to host or a mod in new pre lobby?

**Comment by DanVanAtta** [comment]
@prastle yes
